src/devices/Hx711I2c/Hx711I2c.cs

```diff
@@ -0,0 +1,349 @@
+        /// Otherwise use GetI2cAddress
+        /// </summary>
+        public const int DefaultI2cAddress = 0x64;
+
+        /// <summary>
+        /// Arbitrarily picked value for empty scale.
+        /// </summary>
+        public const float DefaultOffset = 6780606.5f;
+
+        private I2cDevice _i2c;
+
+        /// <summary>
+        /// Raw value telling where 0 is.
+        /// It will be set to current weight when Tare function is used.
+        /// Value passed must be a raw reading.
+        /// </summary>
+        [Property]
+        public float Offset { get; set; }
+
+        /// <summary>
+        /// Value which scales raw units into grams
+        /// </summary>
+        [Property]
+        public float CalibrationScale { get; set; }
+
+        /// <summary>
+        /// When set to true, CAL button will not have any effect on the current calibration setting
+        /// </summary>
+        [Property]
+        public bool IgnoreCalibrationButton { get; set; }
+
+        /// <summary>
+        /// When set to true, RST button will not change Offset (it won't Tare)
+        /// </summary>
+        [Property]
+        public bool IgnoreResetButton { get; set; }
+
+        /// <summary>
+        /// Sets the weight (in grams) used for automatic calibration.
+        /// Value is only relevant when CAL button has been pressed.
+        /// </summary>
+        [Property]
+        public ushort AutomaticCalibrationWeight
+        {
+            // It does not seem to be possible to read from this register
+            set => WriteUInt16(Hx711I2cRegister.REG_SET_TRIGGER_WEIGHT, value);
+        }
+
+        /// <summary>
+        /// Sets the minimum weight in grams which will trigger calibration after CAL button is pressed.
+        /// This value should always be less than calibration weight.
+        /// Value is only relevant when CAL button has been pressed.
+        /// </summary>
+        [Property]
+        public ushort AutomaticCalibrationThreshold
+        {
+            // It does not seem to be possible to read from this register
+            set => WriteUInt16(Hx711I2cRegister.REG_SET_CAL_THRESHOLD, value);
+        }
+
+        /// <summary>
+        /// Hx711I2c - DFRobot KIT0176: I2C 1kg Weight Sensor Kit
+        /// </summary>
+        public Hx711I2c(I2cDevice i2cDevice)
+        {
+            _i2c = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            Offset = DefaultOffset;
+            CalibrationScale = GetAutomaticCalibrationScale();
+        }
+
+        /// <summary>
+        /// Gets I2C address depending on A0 and A1 pin settings.
+        /// </summary>
+        /// <param name="a0">Value of A0 pin.</param>
+        /// <param name="a1">Value of A1 pin.</param>
+        /// <returns>Address of the device.</returns>
+        public static int GetI2cAddress(PinValue a0, PinValue a1)
+            => DefaultI2cAddress + ((bool)a0 ? 1 : 0) + ((bool)a1 ? 2 : 0);
```

Comment:
you should as well present the second I2C address as a constant

Reply:
I think unless I add all 4 of them the second will be confusing while it may be obvious for me that second is `A0=1, A1=0` some other person might thing of second as `A0=1, A1=1` or `A0=0, A1=1`